### ISSUE TITLE
icons position on mate panel

### ISSUE THREAD

**Issue by Stormfinger** [mate-desktop/mate-desktop#560]
#### Expected behaviour
icons should be in order as i put them

#### Actual behaviour
some icons reorder on next boot

#### Steps to reproduce the behaviour
put icon for showing desktop as most right icon beyond clock and calendar icon ..and they switch on next boot

#### MATE general version
mate for debian 1.26

#### Linux Distribution
debian

![MATE-pult](https://github.com/mate-desktop/mate-desktop/assets/84122683/a10fc46b-ccd5-46cc-9269-e2b33839a238)

**Comment by lukefromdc** [comment]
I haven't had thisn issue in years, but I recall stuff like this going on all the way back in the GNOME 2 days. If you are getting it, it means this old bug is back under some condition I have yet to duplicate

**Comment by Stormfinger** [comment]
hi
in this picture is how my panels are organized...i know ...its classic :D 
![panels](https://github.com/mate-desktop/mate-desktop/assets/84122683/bce54d0e-d67a-4b86-85a8-56550bb67b6e)

if i organize icons on left...those app starters then my "show desktop" will move left of "date/time" and if i move "show desktop" as is on picture then my "app starters" will  not be ordered as i want but are in some unknown order

**Comment by lukefromdc** [comment]
One possibility is you may have compressed the applet string on the left when you originally set up the icons. A size change after, a theme change-or an updatge changing the size of anything can then cause an attempt to overlap the icons.  That can cause an applet or launcher to be "kicked out" and show up elsewhere on the panel on a restart or some other conditions.

Unlock the position of everything on the left side of your panel except the menu (first applet) including the window list, temporarily slide them out to the right to make space, then working left to right bring them back to adjacent positions and lock them back down. This should ensure there is always space for them.

**Comment by Stormfinger** [comment]
on left i have brisk menu... i don't know if they still maintain menu i sent them some translations..but to return to panels ...indeed i have changed size of panel to 34 i usually do on every machine ...i do put application panel to middle of the screen and then start adding "app starters" .. then organize them by selecting from menu move and lock them up... then i return application panel next to app starters...on both machines i have debian updated and menus are acting different from machine to machine...sometimes my calendar on boot jumps to most right position sometimes "show desktop" is most right as it should be...

**Comment by crojack** [comment]
I can confirm this is happening to me all the time in Fedora 38 mate-desktop an then I have to unlock all or almost all application lounchers on panel in order to reorder them.

**Comment by lukefromdc** [comment]
Does this happen only with Brisk menu in the panel by any chance?

**Comment by psudo123** [comment]
Genuinely crazy this issue still exists, GNOME 2 was 16 years ago. This needs to be the #1 priority of the MATE team, there is no other issue with the desktop worse than this, I imagine if the team banded together to fix this it could be done in a couple hours.

**Comment by lukefromdc** [comment]
Nobody has ever been able to find the cause of this, or the trigger condition. I suspect this goes all the way
back to the initial release of GNOME 2 and has evaded every attempt to find it. GNOME has and had devs
from Red Hat and even they never were able to fix this. Probablty something to do with applets being able 
to move to accomodate an applet changing width but that covers a lot of ground.

**Comment by vkareh** [comment]
The best solution would be to change applets to use relative positioning rather than the current pixel-based model.

Basically we would define three zones: left/top, center, right/bottom. And the applets get a position index within that zone. That way regardless of panel/screen size, the applets can be re-rendered automatically and always end up in the same relative location.

**Comment by lukefromdc** [comment]
That was tried before and had the effect of trashing prior panel layouts and was buggy enough it was reverted
prior to the 1.28 release

**Comment by suivue** [comment]
> The best solution would be to change applets to use relative positioning rather than the current pixel-based model.
> 
> Basically we would define three zones: left/top, center, right/bottom. And the applets get a position index within that zone. That way regardless of panel/screen size, the applets can be re-rendered automatically and always end up in the same relative location.

Those three zones you are talking about are for mate-panel, correct? If so, Cinnamon panel has this. If it is be implemented, it would be a great addition.

**Comment by vkareh** [comment]
Yes. And I found a whole list of gnome-panel commits that could be used to backport this functionality. I've started an experimental branch to play with that.

**Comment by lukefromdc** [comment]
Just tested the PR, it seemed to work fine
